ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixHealthReportAggregatorTask.java

```diff
@@ -81,16 +86,20 @@ public TaskResult run() {
           statsWrappersJSON.put(instanceName, record.getRecord().getSimpleField(statsFieldName));
         }
       }
-      Pair<String, String> results = clusterAggregator.doWork(statsWrappersJSON, statsReportType);
+      ObjectMapper mapper = new ObjectMapper();
+      Pair<StatsSnapshot, StatsSnapshot> results = clusterAggregator.doWork(statsWrappersJSON, statsReportType);
       String resultId = String.format("Aggregated_%s", healthReportName);
       ZNRecord znRecord = new ZNRecord(resultId);
-      znRecord.setSimpleField(RAW_VALID_SIZE_FIELD_NAME, results.getFirst());
-      znRecord.setSimpleField(VALID_SIZE_FIELD_NAME, results.getSecond());
+      znRecord.setSimpleField(RAW_VALID_SIZE_FIELD_NAME, mapper.writeValueAsString(results.getFirst()));
+      znRecord.setSimpleField(VALID_SIZE_FIELD_NAME, mapper.writeValueAsString(results.getSecond()));
       znRecord.setSimpleField(TIMESTAMP_FIELD_NAME, String.valueOf(SystemTime.getInstance().milliseconds()));
       znRecord.setListField(ERROR_OCCURRED_INSTANCES_FIELD_NAME,
           clusterAggregator.getExceptionOccurredInstances(statsReportType));
       String path = String.format("/%s", resultId);
       manager.getHelixPropertyStore().set(path, znRecord, AccessOption.PERSISTENT);
+      if (statsReportType.equals(StatsReportType.ACCOUNT_REPORT)) {
+        accountService.selectInvalidContainersAndMarkInZK(results.getFirst());
+      }
```

Comment:
This still looks like a synchronous way to call some methods from `accountService`.  Try to initiate another thread to unblock current thread.

Reply:
Inside selectInvalidContainersAndMarkInZK I have two main method, I was trying to start a new thread for markContainerZkNodesInactive, but this update will be include in another PR, so I marked as TODO. But for selectInvalidContainerCandidates() I thought it's ok to run at current thread as we discussed. Please correct me if I'm wrong.

Reply:
OK, that makes sense. I saw the TODO but didn't know how you will implement it. The plan sounds good to me, the remaining thing is, like @zzmao said, whether we should pass in `accountService` to this component. If we decide to decouple them, we can either use a listener or callback.